### PR TITLE
Fix bbox coordinates for ocr_only strategy

### LINKED ISSUE #1227: bug/bbox for pdf gives out nan
When I tried to extract the bbox for a pdf file `layout-parser-paper-fast.pdf` with `ocr_only` strategy, it gives out `nan` values.

```
elements[20].metadata.coordinates
CoordinatesMetadata(points=((nan, nan), (nan, 2200.0), (0.0, 2200.0), (0.0, nan)), 
    system=<unstructured.documents.coordinates.PixelSpace object at 0x2c7b94dc0>)
```

For additional info, the text value of that element is `'2 Z. Shen et al.'`

### PR DESCRIPTION
**Pull request by Coniferish**
### Summary
Closes #1227, which found that `nan` values were present in the coordinates being generated for some elements.
This breaks logic out from `add_pytesseract_bbox_to_elements` to new functions `_get_element_box` and `convert_multiple_coordinates_to_new_system`. It also updates the logic to check that the current bounding box matches the first character of the element's text (as to avoid the `~` characters that `pytesseract.image_to_boxes` includes, but are not present in `pytesseract.image_to_string`.

### Testing
```
from unstructured.partition.image import partition_image
from PIL import Image, ImageDraw

filename="example-docs/layout-parser-paper-with-table.jpg"
elements = partition_image(filename=filename, strategy="ocr_only")
image = Image.open(filename)
draw = ImageDraw.Draw(image)
for i, element in enumerate(elements):
    print(i, element.metadata.coordinates)
    if element.metadata.coordinates:
        draw.polygon(element.metadata.coordinates.points, outline="red", width=2)
output = "example-docs/box-layout-parser-paper-with-table.jpg"
image.save(output)
image.close()
```

### CODE REVIEW

File: unstructured/partition/pdf.py
Comment (badGarnet) [inline unstructured/partition/pdf.py:500]: would this be better if we get word level bounding boxes so it can save us operations? `pytesseract.image_to_data` returns a table with bounding boxes for each "word" (spans bounded by white space). What's more the return can be a pd dataframe so we can do vector match with the table instead of running this in a loop with string operations, type casing, and comparison operations in python (which is more costly than in c++ where tesseract runs)

Comment (Coniferish) [reply unstructured/partition/pdf.py:500]: That sounds like a great idea. Didn't realize pytesseract would return words

File: unstructured/documents/coordinates.py
Comment (badGarnet) [inline unstructured/documents/coordinates.py:71]: nit (aka as a followup ticket): this multiple conversion is just calling single conversion in serial. Looking at the code it seems we might be able to vectorize the operation to improve performance (memory and speed actually; you'd be surprised how much more memory python list takes vs. numpy array).

Comment (Coniferish) [reply unstructured/documents/coordinates.py:71]: I'll create a ticket for this then.

File: unstructured/partition/pdf.py
Comment (Coniferish) [reply unstructured/partition/pdf.py:500]: Should this also be a follow-up issue or should I do this before merging?

File: unstructured/partition/pdf.py
Comment (badGarnet) [inline unstructured/partition/pdf.py:493]: so `char` here might still be not in the `element`. e.g., if the actual text should have been "apple vs. oranges" and `element.text` is `aple vs. oranges`, it is missing the second `p`. Then this logic will still pass because there is still a `p` in the text. More realistically think about a long page it likely has all the most common characters so even if a whole word gets cleaned out the individual characters can still be in `element.txt`. And actually even whole word may still fall into this kind of trap.

File: unstructured/partition/pdf.py
Comment (badGarnet) [reply unstructured/partition/pdf.py:500]: a follow up for sure

File: unstructured/partition/pdf.py
Comment (Coniferish) [reply unstructured/partition/pdf.py:493]: Yah, I realize it's not the best check. It was from when I was initially trying to figure out what to do with the random '~' characters that were showing up. I think this might actually be handled by the `while` loop I later added to compare the first char of element.text and the bounding box char.

Comment (badGarnet) [reply unstructured/partition/pdf.py:493]: so maybe we can remove this logic then?

File: unstructured/partition/pdf.py
Comment (Coniferish) [inline unstructured/partition/pdf.py:631]: @badGarnet Can your # FIXME comment be removed in `_partition_pdf_or_image_with_ocr` with your PR that changed the var names to `_element` and such? Still seeing it in the code and wondering about removing it with this PR

Comment (badGarnet) [reply unstructured/partition/pdf.py:619]: ah; that is regarding to the identical metadata being copied (albeit reference I think) into every element. I think we can remove it for now. I should try to understand its purpose better and propose an alternative if still feel it should be fixed.

File: unstructured/partition/pdf.py
Comment (badGarnet) [inline unstructured/partition/pdf.py:475]:
```suggestion
```

File: unstructured/partition/pdf.py
Comment (badGarnet) [inline unstructured/partition/pdf.py:484]: why can't we just pass in `boxes[box_idx: box_ids+char_count]`? So this function is simplified to iterate through input boxes.

Comment (badGarnet) [inline unstructured/partition/pdf.py:484]: why can't we just pass in `boxes[box_idx: box_ids+char_count]`? So this function is simplified to iterate through input boxes.

File: unstructured/partition/pdf.py
Comment (Coniferish) [reply unstructured/partition/pdf.py:619]: Okay. I'll remove it. It doesn't look to me like there's duplicate metadata being generated, but maybe I'm not looking closely enough.

**Review by badGarnet — APPROVED** [review]

File: unstructured/partition/pdf.py
Comment (qued) [inline unstructured/partition/pdf.py:493]: Should it be `add_pytesseract_bboxes_to_elements`? On the other hand the name `bboxes` is confusing to me for a string, although I'm not familiar with the output of `pytesseract.image_to_boxes`.

**Review by qued — CHANGES_REQUESTED** [review]
Here's some preliminary nits, I'm still trying to understand the changes.

File: unstructured/partition/pdf.py
Comment (Coniferish) [reply unstructured/partition/pdf.py:493]: Will change the name. "bbox" stands for "bounding box", which I saw used as the var name somewhere while working on this

Comment (qued) [reply unstructured/partition/pdf.py]: I understand the abbreviation, but I would expect a variable called `bboxes` to contain a sequence of geometric objects representing bounding boxes... not a string.

Comment (Coniferish) [reply unstructured/partition/pdf.py]: Ahh.. okay. Pytesseract returns a giant string from `image_to_boxes`. Open to recommendations for another name, though.

File: unstructured/partition/pdf.py
Comment (qued) [inline unstructured/partition/pdf.py:470]:
```suggestion
) -> Tuple[Tuple[Tuple[float, float], Tuple[float, int], Tuple[int, int], Tuple[int, float]], int]:
```

File: unstructured/partition/pdf.py
Comment (qued) [reply unstructured/partition/pdf.py]: Ok, how about we just change `bboxes -> bboxes_string`.

File: unstructured/partition/pdf.py
Comment (qued) [inline unstructured/partition/pdf.py:504]:
```suggestion
        bboxes_string (str): The return value of pytesseract.image_to_boxes.
```

**Review by qued — APPROVED** [review]
LGTM pending CI passing.

**Pull request closed without merging**
